src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTestBase.cs

```diff
@@ -193,23 +196,34 @@ public static void CheckNames(IList<MetadataReader> readers, ImmutableArray<Type
         public static void CheckNames(IList<MetadataReader> readers, ImmutableArray<MethodDefinitionHandle> methodHandles, params string[] expectedNames)
             => CheckNames(readers, methodHandles, (reader, handle) => reader.GetMethodDefinition((MethodDefinitionHandle)handle).Name, handle => handle, expectedNames);
 
-        private static void CheckNames<THandle>(
+        internal static void CheckNames<THandle>(
             IList<MetadataReader> readers,
-            ImmutableArray<THandle> entityHandles,
+            IEnumerable<THandle> entityHandles,
             Func<MetadataReader, Handle, StringHandle> getName,
             Func<THandle, Handle> toHandle,
-            string[] expectedNames)
+            string[] expectedNames,
+            string message = null)
         {
+            // If there is only one reader we don't need to aggregate anything
+            if (readers.Count == 1)
+            {
+                MetadataReader reader = readers[0];
+                AssertEx.Equal(expectedNames, entityHandles.Select(handle => reader.GetString(getName(reader, toHandle(handle)))), message: message);
+                return;
+            }
+
             var aggregator = new MetadataAggregator(readers[0], readers.Skip(1).ToArray());
 
             AssertEx.Equal(expectedNames, entityHandles.Select(handle =>
             {
                 var genEntityHandle = aggregator.GetGenerationHandle(toHandle(handle), out int typeGeneration);
                 var nameHandle = getName(readers[typeGeneration], genEntityHandle);
 
-                var genNameHandle = (StringHandle)aggregator.GetGenerationHandle(nameHandle, out int nameGeneration);
-                return readers[nameGeneration].GetString(genNameHandle);
-            }));
+                return readers.GetString(nameHandle);
+
+                //var genNameHandle = (StringHandle)aggregator.GetGenerationHandle(nameHandle, out int nameGeneration);
+                //return readers[nameGeneration].GetString(genNameHandle);
```

Comment:
This should work. Need to investigate if it doesn't

Reply:
What test fails if you keep the original?

Reply:
Oops, wasn't supposed to commit this. I was mucking around here trying to understand why the MetadataAggregator didn't work for the `VerifyTypeDefs` method. Will revert.